azure-iot-device/tests/iothub/test_sync_clients.py

```diff
@@ -69,6 +69,72 @@ def test_sets_on_method_request_received_handler_in_pipleline(
         )
 
 
+class ConfigurationSharedClientCreateFromConnectionStringTests(object):
+    @pytest.mark.it("Sets all configuration options to default when no user configuration provided")
+    def test_pipeline_configuration_defaults(
+        self, mocker, mock_pipeline_init, client_class, connection_string
+    ):
+        mocker.patch("azure.iot.device.iothub.auth.SymmetricKeyAuthenticationProvider")
+
+        mock_config_init = mocker.patch(
+            "azure.iot.device.iothub.abstract_clients.IoTHubPipelineConfig",
+            wraps=config.IoTHubPipelineConfig,
+        )
+
+        args = (connection_string,)
+        client_class.create_from_connection_string(*args)
+
+        assert mock_config_init.call_count == 1
+        assert mock_config_init.call_args == mocker.call()
+        assert mock_pipeline_init.call_args[0][1].websockets is False
+        assert mock_pipeline_init.call_args[0][1].product_info == ""
+
+    @pytest.mark.it("Sets all valid configuration options to the user supplied values")
+    @pytest.mark.parametrize(
+        "websockets, product_info",
+        [
+            pytest.param((None, None), (None, None), id=" Setting to None"),
+            pytest.param(
+                (True, True),
+                ("__fake_product_info__", "__fake_product_info__"),
```

Comment:
>("__fake_product_info__", "__fake_product_info__"), [](start = 16, length = 51)

All this parametrization is overly complex IMHO.

Reply:
But I wouldn't change it.

---
In reply to: [340337876](https://github.com/Azure/azure-iot-sdk-python/pull/368#discussion_r340337876) [](ancestors = 340337876)